### ISSUE TITLE
Cannot recurrent the accuracy

### ISSUE THREAD

**Issue by AyiStar** [tatsuokun/context2vec#2]
I ran the code, using the default config.toml and mscc_train.txt (generated by Holmes_train_data). With n_epochs 5 and 15, I got 0.58 accuracy, much lower than reported.
I want to ask what is wrong, thx.

**Comment by tatsuokun** [comment]
Hi, thank you for reaching out the problem. I just suspect the format of the training dataset.
Does mscc_train.txt  contain one sentence per line?
I got the same result when I didn't permutate the dataset and when the line contained multiple sentences.

**Comment by AyiStar** [comment]
I checked my dataset, and it seems like below.
I think it really contains one sentence per line, right?
And its size is about 220 MB.

it would give us great pleasure to find that gold . ''
`` there 's a boat , with some officers in it , to judge by the amount of gold lace on them , putting off from the ship , '' remarked mr. sharp .
`` ha !
yes !
evidently they intend to pay us a formal visit , '' observed mr. damon .
`` bless my gaiters , though .
i 'm not dressed to receive company .
i think i 'll put on my dress suit . ''
`` it 's too late , '' advised tom .
`` they 'll be here in a minute . ''
urged on by the lusty arms of the brazilian sailors , the boat , containing several officers , neared the floating submarine rapidly .
`` ahoy there ! ''
called an officer in the bow , his accent betraying his unfamiliarity with the english language .
`` what craft are you ? ''
`` submarine , advance , from new jersey , '' replied tom .
`` who are you ? ''
`` brazilian cruiser san paulo , '' was the reply .
`` where are you bound ? ''
went on the officer .
`` on pleasure , '' answered captain weston quickly .
`` but why do you ask ?
we are an american ship , sailing under american colors .
is this brazilian territory ? ''
`` this island is -- yes , '' came back the answer , and by this time the small boat was at the side of the submarine .
before the adventurers could have protested , had they a desire to do so , there were a number of officers and the crew of the san paulo on the small deck .
with a flourish , the officer who had done the questioning drew his sword .
waving it in the air with a dramatic gesture , he exclaimed :
`` you 're our prisoners !
resist and my men shall cut you down like dogs !
seize them , men ! ''
the sailors sprang forward , each one stationing himself at the side of one of our friends , and grasping an arm .
`` what does this mean ? ''
cried captain weston indignantly .
`` if this is a joke , you 're carrying it too far .
if you 're in earnest , let me warn you against interfering with americans ! ''
`` we know what we are doing , '' was the answer from the officer .
the sailor who had hold of captain weston endeavored to secure a tighter grip .
the captain turned suddenly , and seizing the man about the waist , with an exercise of tremendous strength hurled him over his head and into the sea , the man making a great splash .
`` that 's the way i 'll treat any one else who dares lay a hand on me ! ''
shouted the captain , who was transformed from a mild-mannered individual into an angry , modern giant .
there was a gasp of astonishment at his feat , as the ducked sailor crawled back into the small boat .
and he did not again venture on the deck of the submarine .
`` seize them , men ! ''
cried the gold-laced officer again , and this time he and his fellows , including the crew , crowded so closely around tom and his friends that they could do nothing .
even captain weston found it impossible to offer any resistance , for three men grabbed hold of him but his spirit was still a fighting one , and he struggled desperately but uselessly .
`` how dare you do this ? ''
he cried .
`` yes , '' added tom , `` what right have you to interfere with us ? ''
`` every right , '' declared the gold-laced officer .
`` you are in brazilian territory , and i arrest you . ''
`` what for ? ''

**Comment by tatsuokun** [comment]
The dataset seems one-sentence-per-line I believe.
 I just checked those sentences are in my dataset too so I have no idea what's going on.
The reported results are actually averaged on five runs.

**Comment by AyiStar** [comment]
I see, thank you very much !

**Comment by tatsuokun** [comment]
I will check deeply in my code, but I cannot guarantee that I would fix the problem sooner. Guess it takes time from now. If you want to use context2vec, it's faster to use original implementation. Sorry for the inconvenience ;( 
Feel free to send me a pull request if you figure out the solution.

**Comment by tatsuokun** [comment]
My suspicions:
 - Max sentence length; change the number of max_sentences to 128 in main.py.
https://github.com/tatsuokun/context2vec/blob/7bdf8b090582780a6db9339ae1f9df2cf14c1da0/src/main.py#L59

 - Layers of MLP; change the number of the MLP layer to 2 in config.toml.
https://github.com/tatsuokun/context2vec/blob/7bdf8b090582780a6db9339ae1f9df2cf14c1da0/config.toml#L4

Could you send me a result after those changes? I believe it helps you out.

**Comment by AyiStar** [comment]
OK, I'll try, and it will take some time ^_^

**Comment by AyiStar** [comment]
Hi, I tried n_layers=2 and max_sent_length=128, yet n_epochs=5, and got 0.597 accuracy. It's been better slightly, but I think it's not enough. Did you train the model with 5 iterations, too?

**Comment by tatsuokun** [comment]
No, I used 10 iterations.
Since the author of the paper states that 10 iterations can gain accuracy, I also set the same experimental settings to the paper. Please check the original paper for more details.

**Comment by tatsuokun** [comment]
close due to inactivity. Feel free to re-open if you have the same issue.
